Compiling with CryptoAuthLib fails

Compiling *wolfSSL* with *CryptoAuthLib* triggers an ``undefined reference to `atcab_printbin_label'``.

Here is a more complete trace:
```
  CCLD     wolfcrypt/benchmark/benchmark
src/.libs/libwolfssl.so: undefined reference to `atcab_printbin_label'
collect2: error: ld returned 1 exit status
Makefile:3395: recipe for target 'wolfcrypt/benchmark/benchmark' failed
```

I configured *wolfSSL* to use *CryptoAuthLib* using `./configure --with-cryptoauthlib="/home/pi/cryptoauthlib/"`. *CryptoAuthLib* was installed from latest master:
```
git clone https://github.com/MicrochipTech/cryptoauthlib.git && cd cryptoauthlib/lib
cmake . -DATCA_HAL_I2C=ON -DATCA_BUILD_SHARED_LIBS=ON
make
```

I confirm that the above modification solves this issue. However a new `undefined reference` is thrown later:
```
  CCLD     tests/unit.test
tests/tests_unit_test-api.o: In function `test_wc_ecc_verify_hash_ex':
api.c:(.text+0x8938): undefined reference to `wc_ecc_sign_hash_ex'
api.c:(.text+0x8a20): undefined reference to `wc_ecc_sign_hash_ex'
api.c:(.text+0x8a40): undefined reference to `wc_ecc_sign_hash_ex'
api.c:(.text+0x8a60): undefined reference to `wc_ecc_sign_hash_ex'
api.c:(.text+0x8a80): undefined reference to `wc_ecc_sign_hash_ex'
tests/tests_unit_test-api.o:api.c:(.text+0x8aa4): more undefined references to `wc_ecc_sign_hash_ex' follow
```
This happens because [wcc_ecc_sign_hash_ex is not implemented if WOLFSSL_ATECC508A is defined](https://github.com/wolfSSL/wolfssl/blob/0d48a4a1a77126519cb1bd71959e3dfe249165be/wolfcrypt/src/ecc.c#L4441-L4452).

--- a/tests/api.c
+++ b/tests/api.c
@@ -14924,7 +14924,7 @@ static int test_wc_ecc_verify_hash_ex (void)
     int             ret = 0;
 
 #if defined(HAVE_ECC) && defined(HAVE_ECC_SIGN) && defined(WOLFSSL_PUBLIC_MP) \
-     && !defined(WC_NO_RNG)
+     && !defined(WC_NO_RNG) && !defined(WOLFSSL_ATECC508A)
     ecc_key         key;
     WC_RNG          rng;
     mp_int          r;

--- a/wolfcrypt/src/ecc.c
+++ b/wolfcrypt/src/ecc.c
@@ -3979,11 +3979,19 @@ int wc_ecc_make_key_ex(WC_RNG* rng, int keysize, ecc_key* key, int curve_id)
        err = atmel_ecc_create_key(key->slot, key->pubkey_raw);
 
        /* populate key->pubkey */
-       if (err == 0 && key->pubkey.x) {
+       if (err == 0
+       #ifdef ALT_ECC_SIZE
+          && key->pubkey.x
+       #endif
+       ) {
            err = mp_read_unsigned_bin(key->pubkey.x, key->pubkey_raw,
                                       ECC_MAX_CRYPTO_HW_SIZE);
        }
-       if (err == 0 && key->pubkey.y) {
+       if (err == 0
+       #ifdef ALT_ECC_SIZE
+          && key->pubkey.y
+       #endif
+       ) {
            err = mp_read_unsigned_bin(key->pubkey.y,
                                       key->pubkey_raw + ECC_MAX_CRYPTO_HW_SIZE,
                                       ECC_MAX_CRYPTO_HW_SIZE);

--- a/wolfcrypt/src/port/atmel/atmel.c
+++ b/wolfcrypt/src/port/atmel/atmel.c
@@ -105,7 +105,9 @@ int atmel_get_random_number(uint32_t count, uint8_t* rand_out)
 		XMEMCPY(&rand_out[i], rng_buffer, copy_count);
 		i += copy_count;
 	}
+    #ifdef ATCAPRINTF
     atcab_printbin_label((const char*)"\r\nRandom Number", rand_out, count);
+    #endif
 #else
     /* TODO: Use on-board TRNG */
 #endif

--- a/wolfssl/wolfcrypt/include.am
+++ b/wolfssl/wolfcrypt/include.am
@@ -70,7 +70,6 @@ noinst_HEADERS+= \
                          wolfssl/wolfcrypt/port/ti/ti-ccm.h \
                          wolfssl/wolfcrypt/port/nrf51.h \
                          wolfssl/wolfcrypt/port/nxp/ksdk_port.h \
-                         wolfssl/wolfcrypt/port/atmel/atmel.h \
                          wolfssl/wolfcrypt/port/xilinx/xil-sha3.h \
                          wolfssl/wolfcrypt/port/caam/caam_driver.h \
                          wolfssl/wolfcrypt/port/caam/wolfcaam.h \
@@ -78,6 +77,10 @@ noinst_HEADERS+= \
                          wolfssl/wolfcrypt/port/st/stm32.h \
                          wolfssl/wolfcrypt/port/st/stsafe.h \
                          wolfssl/wolfcrypt/port/Espressif/esp32-crypt.h
+
+if BUILD_CRYPTOAUTHLIB
+nobase_include_HEADERS+= wolfssl/wolfcrypt/port/atmel/atmel.h
+endif
 
 if BUILD_AFALG
 nobase_include_HEADERS+= wolfssl/wolfcrypt/port/af_alg/afalg_hash.h

